### PR TITLE
ISSD-2746 [PRM] Improve modal for Deal Registration list

### PR DESCRIPTION
**Pull request by eliasantosf**
Task: [ISSD-2746](https://liferay.atlassian.net/browse/ISSD-2746)

### CODE REVIEW

**Comment by eliasantosf** [comment]
ci:test:sf

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/ModalFormatedInformation/ModalFormatedInformation.tsx [hunk 1]
```diff
@@ -7,22 +7,33 @@ interface ModalFormatedInformationProps {
 	className?: string;
 	information?: string;
 	label: string;
+	sectionTitle?: string;
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/ModalFormatedInformation/ModalFormatedInformation.tsx:10]:
```suggestion
	title?: string;
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/ModalFormatedInformation/ModalFormatedInformation.tsx [hunk 2]
```diff
@@ -7,22 +7,33 @@ interface ModalFormatedInformationProps {
...
 }
 
 const ModalFormatedInformation = ({
 	className = '',
 	information,
 	label,
+	sectionTitle,
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/ModalFormatedInformation/ModalFormatedInformation.tsx:17]:
```suggestion
	title,
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/ModalFormatedInformation/ModalFormatedInformation.tsx [hunk 3]
```diff
@@ -7,22 +7,33 @@ interface ModalFormatedInformationProps {
...
 }: ModalFormatedInformationProps) => (
-	<div className={className}>
-		<h2 className="text-paragraph">{label}</h2>
-
-		{information && (
-			<p className="align-items-center bg-neutral-1 m-0 p-2 rounded text-paragraph-lg">
-				{information}
-			</p>
+	<>
+		{sectionTitle && (
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/ModalFormatedInformation/ModalFormatedInformation.tsx:20]:
```suggestion
		{title && (
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/ModalFormatedInformation/ModalFormatedInformation.tsx [hunk 4]
```diff
@@ -7,22 +7,33 @@ interface ModalFormatedInformationProps {
...
+			<>
+				<div className="mb-0 text-paragraph-md">{sectionTitle}</div>
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/ModalFormatedInformation/ModalFormatedInformation.tsx:22]:
```suggestion
				<div className="mb-0 text-paragraph-md">{title}</div>
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/DealRegistrationList/hooks/useGetListItemsFromDealRegistration.tsx
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/DealRegistrationList/hooks/useGetListItemsFromDealRegistration.tsx:1]: @eliasantosf, those new columns keys were added to table?

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/DealRegistrationList/components/ModalContent/ModalContent.tsx [hunk 5]
```diff
@@ -29,277 +29,190 @@ export default function ModalContent({content, onClose}: ModalContentProps) {
...
+				{content[DealRegistrationColumnKey.PARTNER_ACCOUNT_NAME] && (
+					<ModalFormatedInformation
+						className="d-flex"
+						information={
+							content[
+								DealRegistrationColumnKey.PARTNER_ACCOUNT_NAME
+							]
+						}
+						label="Partner Account Name"
+						sectionTitle="General Information"
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/DealRegistrationList/components/ModalContent/ModalContent.tsx:42]: Instead use a prop, I think it's better to have this Title component here.

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/DealRegistrationList/components/ModalContent/ModalContent.tsx [hunk 6]
```diff
@@ -29,277 +29,190 @@ export default function ModalContent({content, onClose}: ModalContentProps) {
...
+						className="d-flex mb-2"
+						information={
+							content[DealRegistrationColumnKey.PROJECT_TIMELINE]
+						}
+						label="Project Timeline"
+					/>
+				)}
 			</div>
-
 			<div className="d-flex justify-content-end">
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/DealRegistrationList/components/ModalContent/ModalContent.tsx:216]:
```suggestion
			</div>

			<div className="d-flex justify-content-end">
```

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

**Review by jairmedeiros — CHANGES_REQUESTED** [review]
Slight changes!

**Comment by eliasantosf** [comment]
ci:test:sf

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/ModalTitle/ModalTitle.tsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+	title?: string;
+}
+
+const ModalTitle = ({title}: ModalTitleProps) => (
+	<>
+		{title && (
+			<>
+				<div className="mb-0 text-paragraph-md">{title}</div>
+				<hr className="mt-0"></hr>
+			</>
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/ModalTitle/ModalTitle.tsx:16]:
```suggestion
			<div>
				<div className="mb-0 text-paragraph-md">{title}</div>
				<hr className="mt-0"></hr>
			</div>
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/ModalTitle/ModalTitle.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+interface ModalTitleProps {
+	title?: string;
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/ModalTitle/ModalTitle.tsx:7]:
```suggestion
	title: string;
```

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/ModalTitle/ModalTitle.tsx (hunk 7 above)
Comment (eliasantosf) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/ModalTitle/ModalTitle.tsx]: Done!

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by jairmedeiros** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by jairmedeiros** [comment]
ci:forward:force

**Comment by jpince** [comment]
workspaces-compile error in stable:
`     [exec] Exception in thread "main" java.io.IOException: Downloading from https://services.gradle.org/distributions/gradle-7.3.3-bin.zip failed: timeout`
Seems unrelated

**Comment by eliasantosf** [comment]
Related to: https://github.com/brianchandotcom/liferay-portal/pull/147471

**Comment by eliasantosf** [comment]
ci:close

**Pull request closed without merging**